src/Spice86.Core/Emulator/Memory/MemoryRange.cs

```diff
@@ -39,7 +39,7 @@ public static MemoryRange FromSegment(ushort segmentStart, ushort length, string
     /// <returns>A new instance of the <see cref="MemoryRange"/> class.</returns>
     public static MemoryRange FromSegment(ushort segment, ushort startOffset, ushort length, string name) {
         uint start = MemoryUtils.ToPhysicalAddress(segment, startOffset);
-        uint end = MemoryUtils.ToPhysicalAddress(segment, (ushort)(startOffset + length));
```

Comment:
Not so sure if this either introduce a bug or fixes a bug in the calculation of the end address.

What do you think @kevinferrare ?